CSV export can panic on valid data

### Describe the bug

I have some logs that cause VictoriaLogs to panic when I use the query API to export results in CSV format.

(UI doesn't have CSV export, so I use the API)

### To Reproduce

Insert log:
```bash
curl -X POST 'http://127.0.0.1:9428/insert/jsonline?_stream_fields=service,env&_msg_field=message' \
  -H 'Content-Type: application/json' \
  --data-binary "$(printf '%s\n' \
    '{"message":"checkout request for user \"alice@example.com\" failed after upstream replied \"token expired\"","service":"checkout-api","env":"prod","level":"error","request_id":"chk_req_ord_1042_retry_1","route":"/api/v1/orders/checkout","component":"http-server","error_detail":"upstream said \"token expired\" while parsing header \"x-user-id\""}')"
```

Query log
```bash
curl -X POST 'http://127.0.0.1:9428/select/logsql/query' \
  --data-urlencode 'query=* _time:5m' \
  --data-urlencode 'format=csv' \
  -o out.csv
```

### Version

Latest master

### Logs

```
panic: runtime error: slice bounds out of range [:26] with length 10 [recovered, repanicked]

goroutine 52 [running]:
sync.(*WaitGroup).Go.func1.1()
	/Users/xx/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.darwin-arm64/src/sync/waitgroup.go:251 +0x48
panic({0x1037fc040?, 0x133c932383a8?})
	/Users/xx/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.darwin-arm64/src/runtime/panic.go:860 +0x12c
github.com/VictoriaMetrics/VictoriaLogs/app/vlselect/logsql.appendCSVField({0x0, 0x0, 0x0}, {0x133c932bf7a4, 0x5b})
	/Users/xx/logs/app/vlselect/logsql/csv.go:29 +0x344
github.com/VictoriaMetrics/VictoriaLogs/app/vlselect/logsql.appendCSVRow({0x0?, 0x102b287e8?, 0x1037b8420?}, {0x133c93270b08, 0xb, 0x0?}, 0x0)
	/Users/xx/logs/app/vlselect/logsql/logsql.go:1310 +0x6c
github.com/VictoriaMetrics/VictoriaLogs/app/vlselect/logsql.ProcessQueryRequest.func3(0x0, 0x133c9322cd20)
	/Users/xx/logs/app/vlselect/logsql/logsql.go:1286 +0xf8
github.com/VictoriaMetrics/VictoriaLogs/lib/logstorage.(*Storage).RunQuery.WriteDataBlockFunc.newBlockResultWriter.func1(0x0, 0x133c9350a010)
	/Users/xx/logs/lib/logstorage/storage_search.go:186 +0x70
github.com/VictoriaMetrics/VictoriaLogs/lib/logstorage.(*noopPipeProcessor).writeBlock(0x133c9340a4a0, 0x0, 0x133c9350a010)
	/Users/xx/logs/lib/logstorage/pipe.go:110 +0x58
github.com/VictoriaMetrics/VictoriaLogs/lib/logstorage.(*pipeFieldsProcessor).writeBlock(0x133c9323cc48, 0x0, 0x133c9350a010)
	/Users/xx/logs/lib/logstorage/pipe_fields.go:90 +0x68
github.com/VictoriaMetrics/VictoriaLogs/lib/logstorage.(*Storage).searchParallel.func1()
	/Users/xx/logs/lib/logstorage/storage_search.go:1344 +0x1c8
sync.(*WaitGroup).Go.func1()
	/Users/xx/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.darwin-arm64/src/sync/waitgroup.go:258 +0x48
created by sync.(*WaitGroup).Go in goroutine 131
	/Users/xx/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.darwin-arm64/src/sync/waitgroup.go:238 +0x70
make: *** [run] Error 2
```

### Screenshots

_No response_

### Used command-line flags

_No response_

### Additional information

_No response_

--- a/app/vlselect/logsql/csv.go
+++ b/app/vlselect/logsql/csv.go
@@ -40,7 +40,7 @@ func appendCSVField(dst []byte, s string) []byte {
 			break
 		}
 
-		n := strings.IndexByte(s, '"')
+		n = strings.IndexByte(s, '"')
 		if n < 0 {
 			dst = append(dst, s...)
 			break

--- a/app/vlselect/logsql/csv_test.go
+++ b/app/vlselect/logsql/csv_test.go
@@ -25,6 +25,7 @@ func TestAppendCSVField(t *testing.T) {
 	f("\t", "\t")
 
 	f(` foo, bar" baz`, `" foo, bar"" baz"`)
+	f(`foo bar" "baz`, `"foo bar"" ""baz"`) // test multiple quotes
 }
 
 func TestAppendCSVLine(t *testing.T) {

--- a/docs/victorialogs/CHANGELOG.md
+++ b/docs/victorialogs/CHANGELOG.md
@@ -22,7 +22,6 @@ according to the following docs:
 
 ## tip
 
-* BUGFIX: [/select/logsql/query](https://docs.victoriametrics.com/victorialogs/querying/#querying-logs): fix panic when exporting query results in CSV format if a field contains multiple quoted segments. See [#1220](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1220).
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): make legend click perform default action (hide/show series) and move additional actions to hover context menu.
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): improve alert component and simplify error messages for stats view. See [#1128](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1128).
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): use `extra_filters` instead of modifying query when adding legend filters. See [#411](https://github.com/VictoriaMetrics/VictoriaLogs/issues/411).
@@ -38,6 +37,7 @@ according to the following docs:
 * FEATURE: [`/select/logsql/stream_field_names` HTTP endpoint](https://docs.victoriametrics.com/victorialogs/querying/#querying-stream-field-names): add support for an optional `filter=substring` query arg, which allows returning only the field names containing the given `substring`. See [#1186](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1186).
 * FEATURE: [`/select/logsql/stream_field_values` HTTP endpoint](https://docs.victoriametrics.com/victorialogs/querying/#querying-stream-field-values): add support for an optional `filter=substring` query arg, which allows returning only the field values containing the given `substring`. See [#1186](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1186).
 
+* BUGFIX: [/select/logsql/query](https://docs.victoriametrics.com/victorialogs/querying/#querying-logs): fix panic when exporting query results in CSV format if a field contains multiple quoted segments. See [#1220](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1220).
 * BUGFIX: [data ingestion](https://docs.victoriametrics.com/victorialogs/data-ingestion/): properly handle the case when the ingested logs contain `_time` field without the real timestamp, and this field is not mentioned in the `_time_field` query arg or in the `VL-Time-Field` request header according to [these docs](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-parameters). Previously this could lead to unexpected errors during querying such as `missing _time field in the query results`. See [#1168](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1168).
 * BUGFIX: [LogsQL](https://docs.victoriametrics.com/victorialogs/logsql/): properly apply [query options](https://docs.victoriametrics.com/victorialogs/logsql/#query-options) at `vlselect` in [cluster setup](https://docs.victoriametrics.com/victorialogs/cluster/). Previously the options were sent to `vlstorage`, but were ignored at `vlselect`. This could lead to incorrect query results.
 

--- a/docs/victorialogs/CHANGELOG.md
+++ b/docs/victorialogs/CHANGELOG.md
@@ -22,6 +22,7 @@ according to the following docs:
 
 ## tip
 
+* BUGFIX: [/select/logsql/query](https://docs.victoriametrics.com/victorialogs/querying/#querying-logs): fix panic when exporting query results in CSV format if a field contains multiple quoted segments. See [#1220](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1220).
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): make legend click perform default action (hide/show series) and move additional actions to hover context menu.
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): improve alert component and simplify error messages for stats view. See [#1128](https://github.com/VictoriaMetrics/VictoriaLogs/issues/1128).
 * FEATURE: [web UI](https://docs.victoriametrics.com/victorialogs/querying/#web-ui): use `extra_filters` instead of modifying query when adding legend filters. See [#411](https://github.com/VictoriaMetrics/VictoriaLogs/issues/411).

